root/tree/tree/RNTupleLHCBBenchmarks.cxx

```diff
@@ -0,0 +1,102 @@
+#include <ROOT/RNTuple.hxx>
+#include <TH1D.h>
+#include <benchmark/benchmark.h>
+#include <rootbench/RBConfig.h>
+
+namespace {
+constexpr double kKaonMassMeV = 493.677;
+
+double GetP2(double px, double py, double pz)
+{
+   return px * px + py * py + pz * pz;
+}
+
+double GetKE(double px, double py, double pz)
+{
+   double p2 = GetP2(px, py, pz);
+   return sqrt(p2 + kKaonMassMeV * kKaonMassMeV);
+}
+} // namespace
+
+static void BM_RNTuple_LHCB(benchmark::State &state)
+{
+   using RNTupleReader = ROOT::Experimental::RNTupleReader;
+   using RNTupleModel = ROOT::Experimental::RNTupleModel;
+
+   auto model = RNTupleModel::Create();
+   auto ntuple = RNTupleReader::Open(std::move(model), "DecayTree", RB::GetDataDir() + "/B2HHH~none.ntuple");
+   auto viewH1IsMuon = ntuple->GetView<int>("H1_isMuon");
+   auto viewH2IsMuon = ntuple->GetView<int>("H2_isMuon");
+   auto viewH3IsMuon = ntuple->GetView<int>("H3_isMuon");
+
+   auto viewH1PX = ntuple->GetView<double>("H1_PX");
+   auto viewH1PY = ntuple->GetView<double>("H1_PY");
+   auto viewH1PZ = ntuple->GetView<double>("H1_PZ");
+   auto viewH1ProbK = ntuple->GetView<double>("H1_ProbK");
+   auto viewH1ProbPi = ntuple->GetView<double>("H1_ProbPi");
+
+   auto viewH2PX = ntuple->GetView<double>("H2_PX");
+   auto viewH2PY = ntuple->GetView<double>("H2_PY");
+   auto viewH2PZ = ntuple->GetView<double>("H2_PZ");
+   auto viewH2ProbK = ntuple->GetView<double>("H2_ProbK");
+   auto viewH2ProbPi = ntuple->GetView<double>("H2_ProbPi");
+
+   auto viewH3PX = ntuple->GetView<double>("H3_PX");
+   auto viewH3PY = ntuple->GetView<double>("H3_PY");
+   auto viewH3PZ = ntuple->GetView<double>("H3_PZ");
+   auto viewH3ProbK = ntuple->GetView<double>("H3_ProbK");
+   auto viewH3ProbPi = ntuple->GetView<double>("H3_ProbPi");
+
+   auto hMass = new TH1D("B_mass", "", 500, 5050, 5500);
+
+   const Long64_t nEntries = ntuple->GetNEntries() * (state.range(0) / 100.);
+
+   for (auto _ : state) {
+      hMass->Reset();
+      for (auto i = 0ll; i < nEntries; ++i) {
+         ntuple->LoadEntry(i);
```

Comment:
Should be removed as you are using views (the model is empty)

Reply:
done